fcrepo-http-api/src/test/java/org/fcrepo/http/api/services/HttpRdfServiceTest.java

```diff
@@ -0,0 +1,114 @@
+
+import javax.ws.rs.core.MediaType;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.graph.Triple;
+import org.junit.Test;
+import java.io.InputStream;
+import java.io.ByteArrayInputStream;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.runner.RunWith;
+import org.fcrepo.http.commons.api.rdf.HttpIdentifierConverter;
+import org.fcrepo.kernel.api.models.FedoraResource;
+import org.fcrepo.kernel.api.RdfStream;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * Unit tests for HttpRdfService
+ * @author bseeger
+ * @since 2019-11-08
+ */
+@RunWith(MockitoJUnitRunner.Silent.class)
+public class HttpRdfServiceTest {
+
+    @Mock
+    HttpIdentifierConverter idTranslator;
+
+    @Mock
+    FedoraResource resource;
+
+    @InjectMocks
+    private HttpRdfService httpRdfService;
+
+    private final String fedoraUri = "http://www.example.com/fedora/rest/resource1";
+    private final String fedoraId = "info:fedora/resource1";
+    private final String externalUri = "http://www.otherdomain.org/resource5";
+    private final String rdfString = String.format("@prefix dc: <http://purl.org/dc/elements/1.1/> . " +
+                "@prefix dcterms: <http://purl.org/dc/terms/> ." +
+                "<%s>  dc:title 'fancy title' ;" +
+                "  dcterms:isPartOf <%s>, <%s> . ", fedoraUri, externalUri, fedoraUri);
+
+    private final MediaType contentType = new MediaType("text", "turtle");
+    private final InputStream requestBodyStream = new ByteArrayInputStream(rdfString.getBytes());
+
+    @Test
+    public void testGetModelFromInputStream() {
+        when(idTranslator.toInternalId(fedoraUri)).thenReturn(fedoraId);
+        when(idTranslator.inExternalDomain(fedoraUri)).thenReturn(true);
+        when(idTranslator.inExternalDomain(externalUri)).thenReturn(false);
+
+        when(resource.getId()).thenReturn(fedoraId);
+
+        final Model model = httpRdfService.bodyToInternalModel(fedoraUri, requestBodyStream,
+            contentType);
+
+        assertTrue(model.listStatements().toList().size() > 0);
+
+        verifyTriples(model);
+    }
+
+    @Test
+    public void testGetRdfStreamFromInputStream() {
+        when(idTranslator.toInternalId(fedoraUri)).thenReturn("info:fedora/resource1");
+        when(idTranslator.inExternalDomain(fedoraUri)).thenReturn(true);
+        when(idTranslator.inExternalDomain(externalUri)).thenReturn(false);
+
+        final RdfStream stream = httpRdfService.bodyToInternalStream(fedoraUri, requestBodyStream,
+            contentType);
+
+        assertTrue(stream.toString().length() > 0);
+        verifyTriples(stream);
+    }
+
+    private void verifyTriples(final Model model)  {
+        verifyTriples(fromModel(model.getResource(fedoraId).asNode(),model));
+    }
+
+    private void verifyTriples(final RdfStream rdfStream) {
+
+        final List<Triple> triples = rdfStream.map(triple -> {
```

Comment:
This is a bit of an unusual use of the streaming API (there's actually a `peek` method that you can use for doing stuff to items in the stream without consuming them). It might be easier/more readable to transform the stream into a `Model`. I think Jared may have been doing so in one of the tests he was working on. Then you can use nice methods like `model.contains` to check for entire triples at once

Reply:
oh i just realized you convert from a model to a stream. Might be easier to skip that step

Reply:
I think @bseeger is doing this because the actual `bodyToInternalStream` is already outputting a stream so this test will be needed regardless, unless we want to convert the stream to a model in the test. I think it was @dbernstein in his Persister PR #1577 that did the `model.contains`

Reply:
Thanks!!  @whikloj - That's exactly why I did it.  I'll check out @dbernstein's PR.